objc/objc_runtime.go

```diff
@@ -0,0 +1,192 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2022 The Ebitengine Authors
+
+//go:build darwin
+// +build darwin
+
+// Package objc is a low-level pure Go objective-c runtime. This package is easy to use incorrectly, so it is best
+// to use a wrapper that provides the functionality you need in a safer way.
+package objc
+
+import (
+	"fmt"
+	"reflect"
+	"runtime"
+	"unsafe"
+
+	"github.com/ebitengine/purego"
+	"github.com/ebitengine/purego/internal/strings"
+)
+
+//TODO: support try/catch?
+//https://stackoverflow.com/questions/7062599/example-of-how-objective-cs-try-catch-implementation-is-executed-at-runtime
+
+var (
+	objc = purego.Dlopen("/usr/lib/libobjc.A.dylib", purego.RTLD_GLOBAL)
+
+	objc_msgSend           = purego.Dlsym(objc, "objc_msgSend")
+	objc_msgSendSuper2     = purego.Dlsym(objc, "objc_msgSendSuper2")
+	objc_getClass          = purego.Dlsym(objc, "objc_getClass")
+	objc_allocateClassPair = purego.Dlsym(objc, "objc_allocateClassPair")
+	objc_registerClassPair = purego.Dlsym(objc, "objc_registerClassPair")
+	sel_registerName       = purego.Dlsym(objc, "sel_registerName")
+	class_getSuperclass    = purego.Dlsym(objc, "class_getSuperclass")
+	class_addMethod        = purego.Dlsym(objc, "class_addMethod")
+	object_getClass        = purego.Dlsym(objc, "object_getClass")
+)
+
+// Id is an opaque pointer to some Objective-C object
+type Id uintptr
```

Comment:
Let's use `ID` rather than `Id` for the Go's convention (or the Go's standard libraries).